### ISSUE TITLE
`Focus` objects should be serializable to/from lists of dictionaries

### ISSUE THREAD

**Issue by phobson** [pygridgen/pygridgen#41]
Following #40, `Focus` should have `.to_spec()` and `.from_spec(...)` methods.

```python
from pygridgen import grid
fpx = grid._FocusPoint(0.25, 'x', 4, 0.5)
fpy = grid._FocusPoint(0.33, 'y', 0.1, 0.2)
f = grid.Focus(fpx, fpy)
assert f.to_spec() == [
    {'pos': 0.25, 'axis': 'x', 'factor': 4, 'extent': 0.5},
    {'pos': 0.33, 'axis': 'y', 'factor': 0.1, 'extent': 0.2}
]
assert f == grid.Focus.from_spec(f.to_spec())
```

Note, `.from_spec` should be implemented as a `classmethod` like we do [here](https://github.com/Geosyntec/pygridtools/blob/master/pygridtools/core.py#L945)

**Comment by phobson** [comment]
@ricaenriquez, can you look at this and #42?

**Comment by phobson** [comment]
Closed via #47
